redisinsight/api/src/modules/database/database.service.ts

```diff
@@ -197,6 +232,45 @@ export class DatabaseService {
     }
   }
 
+  /**
+   * Test existing database with new fields
+   * @param dto
+   */
+  public async testExistConnection(
+    id: string,
+    dto: PartialDatabaseDto,
+  ): Promise<void> {
+    this.logger.log(`Test exist database connection: ${id}`);
+    const oldDatabase = await this.get(id, true);
+    const database = merge({}, oldDatabase, dto);
+
+    return await this.testConnection(database);
+  }
+
+  /**
+   * Clone database
+   * @param dto
+   */
+  public async clone(
+    id: string,
+    dto: PartialDatabaseDto,
+  ): Promise<Database> {
+    this.logger.log(`Clone database connection: ${id}`);
+    const oldDatabase = omit(await this.get(id, true), 'sshOptions.id');
+    const database = merge({}, omit(oldDatabase, 'id'), dto);
+    if (Object.keys(omitBy(dto, isUndefined)).some((field) => this.connectionFields.includes(field))) {
+      return await this.create(database);
+    }
+
+    const createdDatabase = await this.repository.create({
+      ...classToClass(Database, database),
+      new: true,
+    }, false);
+
```

Comment:
It is by requirements? mark cloned databases as new only when connection fields affected?

Reply:
We marked all cloned databases as new (when connection fields affected we create database and check connection, if connection fields are not affected we will create db instance in repo without check connection) false in 268 line it is a uniqCheck - we don't need this check (only for sso flow)